### PR TITLE
Make batch norm mask shape error more descriptive

### LINKED ISSUE #19818: Error in masked BatchNormalization with > 3 dimensions
```
import keras

x = keras.ops.ones((1, 2, 3, 4))
mask = keras.ops.ones((1, 2), dtype="bool")
y = keras.layers.BatchNormalization()(x, mask=mask, training=True)
```
gives
```
tensorflow.python.framework.errors_impl.InvalidArgumentError: Exception encountered when calling BatchNormalization.call().

{{function_node __wrapped__Mul_device_/job:localhost/replica:0/task:0/device:CPU:0}} Incompatible shapes: [1,2,1] vs. [1,2,3,4] [Op:Mul] name: 

Arguments received by BatchNormalization.call():
  • inputs=tf.Tensor(shape=(1, 2, 3, 4), dtype=float32)
  • training=True
  • mask=tf.Tensor(shape=(1, 2), dtype=bool)
```

### PR DESCRIPTION
**Pull request by pandyah5**
This PR fixes #19818 by adding a more descriptive message to the mask mismatch exception in batch normalization.

Before:
image

After (Highlighted for clarity):
image

### CODE REVIEW

**Comment by gbaned** [comment]
Hi @pandyah5 Can you please sign CLA? Thank you!

**Comment by pandyah5** [comment]
@gbaned I have signed the CLA and rescanned the PR for the check

File: keras/src/layers/normalization/batch_normalization.py [hunk 1]
```diff
@@ -214,6 +214,14 @@ def compute_output_shape(self, input_shape):
         return input_shape
 
     def call(self, inputs, training=None, mask=None):
+        # Check if the mask has one less dimension than the inputs.
+        if mask is not None:
+            if len(mask.shape) != len(inputs.shape) - 1:
+                # Raise a value error
+                raise ValueError(
+                    "The mask provided should be one dimension less than the inputs."
```
Comment (fchollet) [inline keras/src/layers/normalization/batch_normalization.py:222]: Add:

"Received: mask.shape={mask.shape}, inputs.shape={inputs.shape}"

This makes the error easier to debug.

Comment (pandyah5) [reply keras/src/layers/normalization/batch_normalization.py]: Added it! The new error message looks as follows:

image

**Review by fchollet — APPROVED** [review]
LGTM, thanks!

**Pull request merged**